examples/accumulators/template_function_accumulator.hpp

```diff
@@ -106,23 +106,25 @@ namespace examples
         ///          the future should be called. If the value is available,
         ///          get() will return immediately; otherwise, it will block
         ///          until the value is ready.
-        hpx::future<double> query_async()
+        hpx::future<double> query(hpx::launch::async_policy)
         {
             HPX_ASSERT(this->get_id());
 
             typedef server::template_function_accumulator::query_action
                 action_type;
-            return hpx::async<action_type>(this->get_id());
+            return hpx::async<action_type>(hpx::launch::async, this->get_id());
         }
 
         /// Query the current value of the accumulator.
         ///
         /// \note This function is fully synchronous.
-        double query_sync()
+        double query(hpx::launch::sync_policy = hpx::launch::sync)
         {
             HPX_ASSERT(this->get_id());
 
-            return query_async().get();
+            typedef server::template_function_accumulator::query_action
+                action_type;
+            return action_type()(hpx::launch::sync, this->get_id());
```

Comment:
This was possible since the beginnings... After all, it's the equivalent of a normal functions invocation, just look at the table in the SCALA paper you wrote. ;-)